connect/mirror/src/test/java/org/apache/kafka/connect/mirror/MirrorConnectorsIntegrationTest.java

```diff
@@ -150,23 +156,67 @@ public void setup() throws IOException {
         mm2Props.put("backup.bootstrap.servers", backup.kafka().bootstrapServers());
         mm2Config = new MirrorMakerConfig(mm2Props);
 
+        // we wait for the connector and tasks to come up for each connector, so that when we do the
+        // actual testing, we are certain that the tasks are up and running; this will prevent
+        // flaky tests where the connector and tasks didn't start up in time for the tests to be
+        // run
         backup.configureConnector("MirrorSourceConnector", mm2Config.connectorBaseConfig(new SourceAndTarget("primary", "backup"),
             MirrorSourceConnector.class));
+        TestUtils.waitForCondition(() -> assertConnectorAndTasksRunning(backup,
+            "MirrorSourceConnector", MIN_TASKS).isPresent(),
```

Comment:
One more comment that I missed. 

`isPresent` is not perfectly correct here. You'll need to use `orElse(false)`

Basically the idea is that this is successful only if it returns `true`. But if it's `Optional.empty()` or `false` then the assertion is not ready yet.

Reply:
Just for my understanding, why would `isPresent()` not work as well as `orElse(false)`? I thought `isPresent()` only returns `true` when there is a value, and `false` otherwise? Does that not do the same thing as doing `orElse(false)`? I could definitely be missing something here, but just trying to better understand.

Reply:
This assertion tries to capture 3 cases: 
1. All good, return `Optional.of(true)`
2. The boolean logic in the method is `false` (e.g. tasks are not ready, connector is not ready, an error response was returned etc). Please return `Optional.of(false)`
3. Exception was thrown and I can't say. Return `Optional.empty()`

So based on this, how you'll interpret (3) is up to you on the `waitForCondition`. And here you want to treat it as `false`. That's why `orElse(false)` is more appropriate.

Reply:
Ah okay that makes sense. `isPresent()` would only cover the `Optional.empty()` case and would give a false positive if it returned `Optional.of(false)`. `orElse(false)` would make it wait until it's `true`. I'll make the change.

Reply:
After looking at @hachikuji's suggestion, I think it might just be easier (and cleaner) to return the actual boolean instead of an `Optional`.

Reply:
That's fine. There are cases that you might want to invert the answer that says "I can't say". But we haven't finalized the design, that why this helper method hasn't found its way to the base class yet. Still the goal will be to provide generic methods. 

As a reference, here's an example of negation of the above with `orElse(true)`:
https://github.com/apache/kafka/blob/trunk/connect/runtime/src/test/java/org/apache/kafka/connect/integration/RebalanceSourceConnectorsIntegrationTest.java#L227

It's fine to simplify here. We can change if we ever generalize this checks.